stat/roc_test.go

```diff
@@ -256,3 +256,41 @@ func TestROC(t *testing.T) {
 		}
 	}
 }
+
+func TestTOC(t *testing.T) {
+	cases := []struct {
+		c       []bool
+		wantMin []float64
+		wantMax []float64
+		wantTOC []float64
+	}{
+		{ // 0
+			// This is the example given in the paper's supplement.
+			// http://www2.clarku.edu/~rpontius/TOCexample2.xlsx
+			// It is also shown in the WP article.
+			// https://en.wikipedia.org/wiki/Total_operating_characteristic#/media/File:TOC_labeled.png
+			c: []bool{
+				false, true, false, false, true, false,
+				true, false, false, true, true, true,
+				true, true, true, true, false, false,
+				false, false, false, false, false, false,
+				false, false, false, false, false, false,
+			},
+			wantMin: []float64{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
+			wantMax: []float64{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10},
+			wantTOC: []float64{0, 0, 1, 1, 1, 2, 2, 3, 3, 3, 4, 5, 6, 7, 8, 9, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10},
+		},
```

Comment:
I would include tests for all true, all false, len(classes) = 0, etc. similar to ROC

Reply:
Done